pkg/services/ngalert/notifier/compat.go

```diff
@@ -0,0 +1,42 @@
+package notifier
+
+import (
+	"encoding/json"
+
+	alertingNotify "github.com/grafana/alerting/notify"
+
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+)
+
+func PostableGrafanaReceiverToGrafanaReceiver(p *apimodels.PostableGrafanaReceiver) *alertingNotify.GrafanaReceiver {
+	return &alertingNotify.GrafanaReceiver{
+		UID:                   p.UID,
+		Name:                  p.Name,
+		Type:                  p.Type,
+		DisableResolveMessage: p.DisableResolveMessage,
+		Settings:              json.RawMessage(p.Settings),
+		SecureSettings:        p.SecureSettings,
+	}
+}
+
+func PostableApiReceiverToApiReceiver(r *apimodels.PostableApiReceiver) *alertingNotify.APIReceiver {
+	receivers := alertingNotify.GrafanaReceivers{
+		Receivers: make([]*alertingNotify.GrafanaReceiver, 0, len(r.GrafanaManagedReceivers)),
+	}
+	for _, receiver := range r.GrafanaManagedReceivers {
+		receivers.Receivers = append(receivers.Receivers, PostableGrafanaReceiverToGrafanaReceiver(receiver))
+	}
+
+	return &alertingNotify.APIReceiver{
+		ConfigReceiver:   r.Receiver,
+		GrafanaReceivers: receivers,
+	}
+}
+
+func PostableApiAlertingConfigToApiReceivers(c apimodels.PostableApiAlertingConfig) []*alertingNotify.APIReceiver {
+	apiReceivers := make([]*alertingNotify.APIReceiver, len(c.Receivers))
```

Comment:
I think this is a bug and the start of the slice will have nil pointers due to `append`:
```suggestion
	apiReceivers := make([]*alertingNotify.APIReceiver, 0, len(c.Receivers))
```

I think it would be good to make sure these functions have tests 🙂